src/tools/networkTools.cpp

```diff
@@ -211,40 +217,35 @@ std::map<std::string, std::string> getNetworkInterfaces() {
   return result;
 }
 
-
-std::string getBestPublicIp(bool ipv6) {
-  IpAddress bestPublicIp = IpAddress{"127.0.0.1","::1"};
+IpAddress getBestPublicIps() {
+  IpAddress bestPublicIps;
   std::map<std::string, IpAddress> interfaces = getNetworkInterfacesIPv4Or6();
-
 #ifndef _WIN32
-  const char* const prioritizedNames[] =
-      { "eth0", "eth1", "wlan0", "wlan1", "en0", "en1" };
-  for(auto name: prioritizedNames) {
-    auto it=interfaces.find(name);
-    if(it != interfaces.end() && !(ipv6 && (*it).second.addr6.empty())) {
-      bestPublicIp = (*it).second;
-      break;
+  const char* const prioritizedNames[] = { "eth0", "eth1", "wlan0", "wlan1", "en0", "en1" };
+  for (const auto& name : prioritizedNames) {
+    const auto it = interfaces.find(name);
+    if (it != interfaces.end()) {
+      updatePublicIpAddress(bestPublicIps, it->second);
     }
   }
 #endif
-
-  const char* const prefixes[] = { "192.168", "172.16.", "10.0" };
-  for(auto prefix : prefixes){
-    for(auto& itr : interfaces) {
-      std::string interfaceIp(itr.second.addr);
-      if (interfaceIp.find(prefix) == 0 && !(ipv6 && itr.second.addr6.empty())) {
-        bestPublicIp = itr.second;
-        break;
+  const char* const v4prefixes[] = { "192.168", "172.16", "10.0", "169.254" };
```

Comment:
Adding the 169.254 prefix may have been a mistake. While testing the fix for https://github.com/kiwix/libkiwix/issues/1144 I ran against a situation when the public API displayed by kiwix-serve starts with this prefix but that address doesn't work.

Reply:
Not sure what this range actually means?

Reply:
@sgourdas Can you please try to reproduce the issue? For me it occurred under Windows.

Reply:
@veloman-yunkan I confirm that this is an issue I run into as well.

Reply:
> Not sure what this range actually means?

@kelson42 It's a link-local address. Discussed here https://github.com/kiwix/libkiwix/pull/1132#discussion_r1774806791

Reply:
@sgourdas Please open asap an issue

Reply:
> @veloman-yunkan I confirm that this is an issue I run into as well.

@sgourdas Then please file an issue and let's fix it before libkiwix v14.0 is released.